### ISSUE TITLE
Still ongoing?

### ISSUE THREAD

**Issue by pilotbaba** [Interactml/iml-unity#6]
Seems like some awesome work invested in this tool. It would be a shame not to pursue further development. I am eager to use it in my research but I am apprehensive on the usability of the plugin in subsequent versions of Unity. Further details/documentation on ML specifics would also be nice... So, I wish you a productive 2022 :)

**Comment by carlotes247** [comment]
Thank you for the praise! Development is still ongoing, but I am right now wrapping up a paper that needs to be submitted this week. After that, the plan is to have a new InteractML version ready for the upcoming Global Game Jam at the end of January.

I haven't tried with new versions of Unity, but InteractML should work as it is as long as they keep supporting the traditional EditorGUI. What kind of docs would you find useful about the underlying ML algorithms by the way? I am more than happy to extend the current section of the wiki :)

**Comment by pilotbaba** [comment]
Many thanks for your reply. Well basically the under-the-hood info on type of regression being computed, what sort of classification is used and the libraries you rely on. This is merely for reporting purposes in any future publication I might have with using this plugin. Other than this the plugin is concise and you are targeting a very specific niche that is otherwise vacant in Unity at least. 
I would be interested in the inclusion of real-time, training-free comparison of two samples of data using DTW. Basically my priority is to have an open-ended way of comparing e.g., motion without aiming for imitation of any pre-trained gesture. Is there a way that I am missing to achieve this somehow already?

**Comment by carlotes247** [comment]
Hi, sorry for taking a while to get back. The paper is now out of my to-do list :) InteractML uses [rapidlib](https://github.com/mzed/RapidLib)'s kNN, MLP and DTW implementation in C++ . From our latest [publication](https://dl.acm.org/doi/10.1145/3489849.3489879): 

> "In InteractML, regression is implemented using a multi-layer perceptron neural network with one hidden layer; such an algorithm does not place undue restrictions on the shape of the model function to be learned (e.g., functions can be nonlinear) yet can often be trained effectively on as few as a dozen examples."

**Comment by carlotes247** [comment]
I just added some more details on the readme file of the project under 'Machine Learning Algorithms'. Could you elaborate a bit more on what do you mean with the real-time, training-free comparison of the two DTW samples?

**Comment by pilotbaba** [comment]
I formulated the question badly but in the meantime I believe I figured out that what I want is out of the scope of your work since it does not involve machine learning in any form. What I am looking for is basically an implementation of DTW in Unity to be used to compare cached signals, e.g. one would be obtained from a dance instructor and another from the pupil trying to imitate. So there would be no comparison with any model, just a quantification of a measure of their match in movement.
